server/services/team/service.ts

```diff
@@ -159,4 +167,40 @@ export class TeamService extends ServiceClass implements ITeamService {
 
 		return this.TeamMembersModel.findByTeamId(teamId).toArray();
 	}
+
+	async addMember({ _id, username }: IUser, userId: string, teamId: string): Promise<InsertOneWriteOpResult<ITeamMember> | null | undefined> {
+		const isAlreadyAMember = await this.TeamMembersModel.findOneByUserIdAndTeamId(userId, teamId, { projection: { _id: 1 } });
+
+		if (isAlreadyAMember) {
+			return;
+		}
+
+		return this.TeamMembersModel.createOneByTeamIdAndUserId(teamId, userId, { _id, username });
+	}
+
+	async getOneByRoomId(roomId: string): Promise<ITeam | null> {
+		return this.TeamModel.findOneByMainRoomId(roomId, { projection: { _id: 1 } });
+	}
+
+	async addRolesToMember(teamId: string, userId: string, roles: Array<string>): Promise<UpdateWriteOpResult | undefined> {
```

Comment:
I'd avoid using the return of mongodb driver on the service as it can have complex objects that might not be able to be transferred via network. for this action I'd return a `boolean`

Reply:
Done!